src/lib/components/MainPanel.js

```diff
@@ -66,18 +61,12 @@ export default class MainPanel extends React.Component {
     scheduleSnoozedTab(time);
   }
 
-  handleTimeSelect(value) {
-    this.setState({ dateChoice: value });
-  }
-
   closeTimeSelect() {
     this.setState({ datepickerActive: false });
   }
 
-  confirmTimeSelect() {
-    const { dateChoice } = this.state;
+  confirmTimeSelect(dateChoice) {
     const { scheduleSnoozedTab } = this.props;
-
     if (!dateChoice) { return; }
```

Comment:
(It also looks like setting an alarm for a time in the past doesn't ever fire the trigger, so we should _really_ check for that before setting the alarm.)